src/server/frontend_wayland/xdg_activation_v1.cpp

```diff
@@ -273,8 +285,13 @@ void mf::XdgActivationTokenV1::commit()
             return;
         }
 
-        // TODO:
-        // Next, verify that the serial belongs to the seat
+        auto event = client->event_for(serial.value());
+        if (!event)
+        {
+            mir::log_error("XdgActivationTokenV1::commit serial not found");
```

Comment:
I think this is at most a "warning" and possibly an "info": this doesn't impact the running of the server, just indicates the client is misbehaving. (If possible, it might also be helpful to indicate which client)